### DISCUSSION TITLE
Correct stress in words

### DISCUSSION THREAD

**Discussion by Karlanta** [DrewThomasson/ebook2audiobook#837] [Q&A] [answered]
Hello! How do I put the right accents in words? The AI pronounces unfamiliar words (invented names in a novel or existing complex words) incorrectly. The methods that are suitable for other neural networks (capital letters, the + sign before the stressed vowel letter) do not work.

**Comment by ROBERT-MCDOWELL** [comment]
you must provide more informations of your eb2ab cmd + settings + language, system, native docker etc....

**Reply by Karlanta** [reply]
Do you mean that there is no way for this neural network to technically format a text file with the correct accents? Because any settings will not affect the pronunciation of previously non-existent names and toponyms that I invented for my own book... 
I will try to give information about the system. Ebook2Audiobook v25.6.9 native mode, without Docker, Russian language. I generate on a CPU of the 13th generation Intel i5-13500 2.50 GHz, Windows 10 Professional 64x. I clone Russian voices using the GUI, the samples are about 10 minutes each. As a result, 90% of the accents generated are correct, 10% are not.  Morgan Freeman's voice reads the same text the same way, and maybe even a little better, but it still makes mistakes in the accents of rare words, the same 5-10% of all fragments. (Хрипота, рукоприкладство, оленья - all this is pronounced with the wrong emphasis)

![image](https://github.com/user-attachments/assets/e91b6053-3704-4c71-9c8f-3bfed003f34d)

I don't understand what means eb2ab cmd, all the process? Here it is.

Processing eBook file: 1.txt
G:\EBOOK2AUDIO\ebook2audiobook\python_env\Lib\site-packages\iso639\iso639.py:20: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  from pkg_resources import resource_filename
Using CPU -
Running command: C:\Users\BROMDEN\scoop\shims\ebook-convert.EXE G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\1.txt G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\__1.epub
Conversion options changed from defaults:
  input_encoding: 'utf-8'
  verbose: 1
  epub_version: '3'
  smarten_punctuation: True
  disable_font_rescaling: True
  output_profile: 'generic_eink'
  flow_size: 0
1% Конвертирую входной файл в HTML...
InputFormatPlugin: TXT Input running
on G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\1.txt
Reading text from file...
Using user specified input encoding of utf-8
Auto detected paragraph type as single
Auto detected formatting as heuristic
Running text through basic conversion...
Language not specified
Creator not specified
Building file list...
        Found files...
                 HTMLFile:0:a:'G:\\EBOOK2AUDIO\\ebook2audiobook\\tmp\\calibre-teero96y\\co_1lbfp_plumber\\index.html'
Normalizing filename cases
Rewriting HTML links
Parsing index.html ...
*********  Heuristic processing HTML  *********
There are 0 blank lines. 0.0 percent blank
minimum chapters required are: 1
found 0 pre-existing headings
Total wordcount is: 110, Average words per section is: 110, Marked up 0 chapters
Hard line breaks check returned False
Median line length is 13, calculated with html format
Fixing hyphenated content
Looking for more split points based on punctuation, currently have 0
Formatting scene breaks
Forcing index.html into XHTML namespace
34% Выполняются изменения в электронной книге…
Merging user specified metadata...
Detecting structure...
Auto generated TOC with 0 entries.
Flattening CSS and remapping font sizes...
Source base font size is 12.00000pt
Removing fake margins...
Found 5 items of level: p_1
Ignoring level p_1
Cleaning up manifest...
Trimming unused files from manifest...
Creating EPUB Output...
67% Запускается плагин EPUB Output
Splitting markup on page breaks and flow limits, if any...
Generating default cover
This EPUB file has no Table of Contents. Creating a default TOC
Upgrading to EPUB 3...
EPUB output written to G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\__1.epub
Выходные данные сохранены в   G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\__1.epub


*******************************************************************************
NOTE:
The warning "Character xx not found in the vocabulary."
MEANS THE MODEL CANNOT INTERPRET THE CHARACTER AND WILL MAYBE GENERATE
(AS WELL AS WRONG PUNCTUATION POSITION) AN HALLUCINATION TO IMPROVE THIS MODEL,
IT NEEDS TO ADD THIS CHARACTER INTO A NEW TRAINING MODEL.
YOU CAN IMPROVE IT OR ASK TO A TRAINING MODEL EXPERT.
*******************************************************************************

G:\EBOOK2AUDIO\ebook2audiobook\lib\classes\tts_engines\coqui.py:68: UserWarning: Environment variable TORCH_FORCE_NO_WEIGHTS_ONLY_LOAD detected, since the`weights_only` argument was not explicitly passed to `torch.load`, forcing weights_only=False.
  xtts_builtin_speakers_list = torch.load(self.speakers_path)
Loading TTS xtts model, it takes a while, please be patient...
xtts Loaded!
conversion 0.00%: 0/5 Block 1 containing 5 sentences...
Computing speaker latents...
Converting 0.00%: : 0/5
Sentence: А скорей всего - к Лотпранду , они всегда ладили . ‡pause‡- Неужели все настолько плохо ?
Converting 20.00%: : 1/5
Sentence: ‡pause‡- Настолько . ‡pause‡Фронадан замолчал , не желая сплетничать о том , чему стал свидетелем два года назад , когда все Саврайсы собрались в Хаубере на Большую оленью охоту .
Converting 40.00%: : 2/5
Sentence: Видит Единый , бывало , он и сам спорил с отцом до хрипоты , но дело никогда не доходило до рукоприкладства . Сцена же между Ганаханом и его отцом была неприглядной .
Converting 60.00%: : 3/5
Sentence: ‡pause‡Фронадан шел тогда к покоям Адемара , но по пути услышал из-за приоткрытой двери комнат Сейтера его громкий голос и тонкий голосок Ганахана . Они яростно спорили .
Converting 80.00%: : 4/5
Sentence: Уже почти пройдя мимо , он вдруг услышал звонкий щелчок оплеухи и увидел через щель , как мальчик катится по полу .
Converting 80.00%: : 5/5 End of Block 1
Input #0, concat, from 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\chapters\sentences\sentences.txt':
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Audio: flac, 24000 Hz, mono, s16
Stream mapping:
  Stream #0:0 -> #0:0 (flac (native) -> flac (native))
Press [q] to stop, [?] for help
Output #0, flac, to 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\chapters\chapter_1.flac':
  Metadata:
    encoder         : Lavf61.7.100
  Stream #0:0: Audio: flac, 24000 Hz, mono, s16, 128 kb/s
      Metadata:
        encoder         : Lavc61.19.101 flac
[out#0/flac @ 0000027647329c80] video:0KiB audio:1210KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 0.668697%
size=    1218KiB time=00:00:51.11 bitrate= 195.2kbits/s speed=1.11e+03x
********* Combined block audio file saved to G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\chapters\chapter_1.flac
Combining block 1 to audio, sentence 0 to 4
Converting 80.00%: : 5/5
Input #0, concat, from 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\chapters\chapters.txt':
  Duration: N/A, start: 0.000000, bitrate: N/A
  Stream #0:0: Audio: flac, 24000 Hz, mono, s16
Stream mapping:
  Stream #0:0 -> #0:0 (flac (native) -> flac (native))
Press [q] to stop, [?] for help
Output #0, flac, to 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\1.flac':
  Metadata:
    encoder         : Lavf61.7.100
  Stream #0:0: Audio: flac, 24000 Hz, mono, s16, 128 kb/s
      Metadata:
        encoder         : Lavc61.19.101 flac
[out#0/flac @ 000001cf3aa3d900] video:0KiB audio:1210KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 0.668697%
size=    1218KiB time=00:00:51.11 bitrate= 195.2kbits/s speed=1.99e+03x
********* total audio blocks saved to G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\1.flac
Input #0, flac, from 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\1.flac':
  Metadata:
    encoder         : Lavf61.7.100
  Duration: 00:00:51.11, start: 0.000000, bitrate: 195 kb/s
  Stream #0:0: Audio: flac, 24000 Hz, mono, s16
Input #1, ffmetadata, from 'G:\EBOOK2AUDIO\ebook2audiobook\tmp\ebook-5e58dc14-9e9e-4387-8185-f01ca578cb16\6f331d4760b8edd29267b7891d24c56a\metadata.txt':
  Metadata:
    title           : 1
    artist          : Неизвестный
    year            : 2025
  Duration: 00:00:51.12, start: 0.000000, bitrate: 0 kb/s
  Chapters:
    Chapter #1:0: start 0.000000, end 51.115000
      Metadata:
        title           : Part 1
Stream mapping:
  Stream #0:0 -> #0:0 (flac (native) -> pcm_s16le (native))
Press [q] to stop, [?] for help
Output #0, wav, to 'G:\EBOOK2AUDIO\ebook2audiobook\audiobooks\gui\host\web-5e58dc14-9e9e-4387-8185-f01ca578cb16\1.wav':
  Metadata:
    INAM            : 1
    IART            : Неизвестный
    ILNG            : rus
    year            : 2025
    ISFT            : Lavf61.7.100
  Chapters:
    Chapter #0:0: start 0.000000, end 51.115000
      Metadata:
        title           : Part 1
  Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 24000 Hz, mono, s16, 384 kb/s
      Metadata:
        encoder         : Lavc61.19.101 pcm_s16le
[out#0/wav @ 000001cdebbe8900] video:0KiB audio:2396KiB subtitle:0KiB other streams:0KiB global headers:0KiB muxing overhead: 0.005380%
size=    2396KiB time=00:00:51.11 bitrate= 384.0kbits/s speed=3.69e+03x

*********** Session: 5e58dc14-9e9e-4387-8185-f01ca578cb16 **************
Store it in case of interruption, crash, reuse of custom model or custom voice,
you can resume the conversion with --session option

**Reply by SOVA001** [reply]
Hello, like you, I tried to voice a book in Russian, I would like to know which voices are best for Russian books. Which voice do you think is better? Are you using the built-in voice or have you uploaded a custom one?

**Comment by ROBERT-MCDOWELL** [comment]
pronouciation comes from the model itself, If you wish another pronounciation and create new words or else that you are expecting for so the solution is to create your own fine tuned model.

**Reply by Karlanta** [reply]
Thanks for the reply, I get it, I'll work on it.

**Comment by DrewThomasson** [comment, marked as answer]
looks like this was something that they never added to xtts :/ 

[SSML] Manual Stress Control in xtts

https://github.com/coqui-ai/TTS/issues/3039

the best I can say is to fine-tune it or find a Phonetic spelling of it that makes xtts or bark pronounce it correctly
heres a free hugginface demo space for xtts to mess around in
https://huggingface.co/spaces/coqui/xtts

**Reply by Karlanta** [reply]
Thanks for the reply, I'll try to figure it out.
